text-serializer-gson/src/main/java/net/kyori/adventure/text/serializer/gson/StyleSerializer.java

```diff
@@ -99,6 +110,9 @@ static TypeAdapter<Style> create(final net.kyori.adventure.text.serializer.json.
   private final net.kyori.adventure.text.serializer.json.LegacyHoverEventSerializer legacyHover;
   private final boolean emitLegacyHover;
   private final boolean emitModernHover;
+  private final boolean emitSuperModernHover;
```

Comment:
I'm not sure if `superModern` is a right fit here. What if Mojang introduced further changes, would it be `superDuperModern`?

Reply:
ideally end users should just use something like 
```
GsonComponentSerializer.builder()
              .options(JSONOptions.byDataVersion().at(Bukkit.getUnsafe().getDataVersion()))
              .build();
```
anyways

Reply:
pushed something, how does that feel?

Reply:
It's way better